app/controllers/slots_controller.rb

```diff
@@ -0,0 +1,90 @@
+class SlotsController < ApplicationController
+  load_and_authorize_resource
+  def index
+    @slots = Slot.where(booked: false).where("available_days >= ?", Date.today).order(:available_days, :available_time).map.with_index(1) do |slot, index|
+      start_time = slot.available_time.strftime("%H:%M")
+      end_time = (slot.available_time + 1.hour).strftime("%H:%M")
+      available_days = slot.available_days.strftime('%d-%m-%y')
+      user_name = "#{slot.user.first_name} #{slot.user.last_name}" if slot.user
+      {
+        "slot_id": slot.id,
+        "Doctor_name": user_name,
+        "available_date": "#{available_days}",
+        "slot_#{index}": " #{start_time} to #{end_time}"
+      }
+    end
+    render json: @slots
+  end
+    
+  def create
+    available_time = Time.zone.parse(slot_params[:available_time])
+    if available_time.hour < 10 || available_time.hour >= 18
+      render json: { error: "Slots must start between 10:00 and 17:00" }, status: :unprocessable_entity
```

Comment:
@Smokashi23 Please review your all files in this PR and change messages through I18n